app/uk/gov/hmrc/minorentityidentificationfrontend/controllers/trustControllers/CaptureCHRNumberController.scala

```diff
@@ -0,0 +1,102 @@
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.minorentityidentificationfrontend.controllers.trustControllers
+
+import play.api.mvc.{Action, AnyContent, MessagesControllerComponents}
+import uk.gov.hmrc.auth.core.retrieve.v2.Retrievals.internalId
+import uk.gov.hmrc.auth.core.{AuthConnector, AuthorisedFunctions}
+import uk.gov.hmrc.http.InternalServerException
+import uk.gov.hmrc.minorentityidentificationfrontend.config.AppConfig
+import uk.gov.hmrc.minorentityidentificationfrontend.controllers.overseasControllers.{routes => overseasControllersRoutes}
+import uk.gov.hmrc.minorentityidentificationfrontend.controllers.trustControllers.{routes => trustControllersRoutes}
+import uk.gov.hmrc.minorentityidentificationfrontend.featureswitch.core.config.{EnableFullTrustJourney, FeatureSwitching}
+import uk.gov.hmrc.minorentityidentificationfrontend.forms.trustForms.CaptureCHRNForm
+import uk.gov.hmrc.minorentityidentificationfrontend.services.{JourneyService, StorageService}
+import uk.gov.hmrc.minorentityidentificationfrontend.views.html.trustViews.capture_chrn_page
+import uk.gov.hmrc.play.bootstrap.frontend.controller.FrontendController
+
+import javax.inject.{Inject, Singleton}
+import scala.concurrent.ExecutionContext
+
+@Singleton
+class CaptureCHRNumberController @Inject() (val authConnector: AuthConnector,
+                                           journeyService: JourneyService,
+                                           storageService: StorageService,
+                                           mcc: MessagesControllerComponents,
+                                           view: capture_chrn_page)
+                                           (implicit val config: AppConfig, executionContext: ExecutionContext)
+                                           extends FrontendController(mcc) with AuthorisedFunctions with FeatureSwitching {
+
+  def show(journeyId: String): Action[AnyContent] = Action.async {
+    implicit request =>
+      authorised().retrieve(internalId) {
+        case Some(authInternalId) =>
+          if(isEnabled(EnableFullTrustJourney)) {
+            journeyService.getJourneyConfig(journeyId, authInternalId).map {
+              journeyConfig => Ok(view(
+                journeyId = journeyId,
+                pageConfig = journeyConfig.pageConfig,
+                formAction = trustControllersRoutes.CaptureCHRNumberController.submit(journeyId),
+                form = CaptureCHRNForm.form
+              ))
+            }
+          } else throw new InternalServerException("Trust journey is not enabled")
+        case None =>
+          throw new InternalServerException("Internal ID could not be retrieved from Auth")
+      }
+  }
+
+  def submit(journeyId: String): Action[AnyContent] = Action.async {
+    implicit request =>
+      authorised().retrieve(internalId) {
+        case Some(authInternalId) =>
+          if(isEnabled(EnableFullTrustJourney)) {
```

Comment:
Could we please put this clause on the other new pages to check if the FS is enabled?

Reply:
I'll create a new ticket for this.